src/parthenon_arrays.hpp

```diff
@@ -144,6 +143,12 @@ class ParArrayNDGeneric {
     Kokkos::deep_copy(d6d_, src.Get());
   }
 
+  auto GetHostMirrorAndCopy() {
+    auto host_mirror = GetHostMirror();
+    Kokkos::deep_copy(host_mirror.Get(), d6d_);
+    return host_mirror;
+  }
+
```

Comment:
`GetHostMirror` returns a `ParArrayND`, not a mirror view. So if you want to mimic the `ParArrayND` interface, you need to not use `Kokkos::deep_copy`. Also you should probably use `Kokkos::create_mirror_view_and_copy`, which (I think) will be more efficient than `create_mirror_view` and then `deep_copy`. @dholladay00 might be able to correct me.

Summarizing, it should look like:

```C++
template<typename MemSpace>
auto GetMirrorAndCopy(MemSpace const &memspace) {
  auto mirror = Kokkos::create_mirror_view_and_copy(memspace, d6d_);
  return ParArrayNDGeneric<decltype(mirror)>(mirror);
}
auto GetHostMirrorAndCopy() { return GetMirrorAndCopy(Kokkos::HostSpace()); }
auto GetDeviceMirrorAndCopy() { return GetMirrorAndCopy(Kokkos::DefaultExecutionSpace()); }
```

Reply:
I think it's cleaner, but I'm not sure that it's any more efficient since deep_copy will no-op if the views alias the same data. Its primary benefit from my perspective is when appending traits to a view. For example, if I have some inputs that are const and being passed in, I can do `View<const T**, Layout, DevMemSpace, RandomAccess> inputs = create_mirror_view_and_copy(DevMemSpace, inputs_host)` rather than having to create a non-const view, deep_copy into it, and then assign to the const random access view.

Reply:
Hmm. Well that's probably a reason to use `create_mirror_view_and_copy` since `ParArrayND` can in principle carry around traits.

Reply:
I see. Thanks for those suggestions. I'll update accordingly.